graphium/trainer/losses.py

```diff
@@ -63,6 +63,8 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
         # regression loss needs normalized logits to probability as input to do inner product with self.brackets
         # we apply softmax on the raw logits first
         softmax_input = self.softmax(input)
+        # the softmax of a tensor of 0s would not be 0s anymore, so need to apply nan_targets here again to filter out
+        softmax_input[nan_targets] = 0.0
```

Comment:
Found a bug here: the nans are set to 0s in `ipu_losses.py` and here `softmax` is applied, while the `softmax` of 0s are not 0s. Previous approach would lead to wrong estimation of hybridceloss.